### PR TITLE
tooltip/popover: add a `customClass` option

### LINKED ISSUE #19415: Simplify the customization of popover
This is a suggestion I have from a long time. I thought it was missing in version 3, and it might be well to allow it in version 4.
When we want to customize the popover (width, content padding...) we actually do not have other choice that to redefine the full template, which is not very readable, maintainable and evolutive.
A better option might be to allow to add a class to the popver, so we could add our custom styles to this class without impacting all the popovers.

Example :

```
.popover-custom .popover-content {
  padding: 0;
}

$('#foo').popover({
  title: ...,
  content: ...,
  class: 'popover-custom'
});
```

### PR DESCRIPTION
**Pull request by remeika**
**Please squash before merging**

Fixes #19415

Add extra classes to Tooltips and Popovers using the `additionalClasses` option, using [`$.addClass()`](https://api.jquery.com/addClass/) behind the scenes. 

Details:

- Additional classes are evaluated and added every time the `show()` command is run. This should be idempotent, and allows the class list to be updated every time the component is shown.
- New classes are added after the component template is hydrated in the DOM; if classes are added to a custom template and using `additionalClasses`, both will show up. 

Questions I still have:

- I targeted this release at the 4.X branch. I am not sure if I did the right thing there, rather than basing my work off `main`.
- It looks like the file size of `bootstrap.bundle.js` has been locked in place: adding any additional code causes the build pipeline to fail. I'm wondering what course of action an individual contributor should take in this situation: Is the 4.X branch feature-locked? Should I try to delete some code elsewhere to try to squeeze under the limit? Should I just raise the limit?

### CODE REVIEW

**Comment by XhmikosR** [comment]
Thanks for the PR!

That being said, we always target the main branch first and then we backport anything if deemed necessary. That being said, since the tests differ, we'd need a v4-dev patch anyway.

But we'd need a patch for the main branch anyway. About bundlewatch, leave that to us, we'll update the file before merging.

File: js/src/tooltip.js
Comment (XhmikosR) [inline js/src/tooltip.js:445]: TBH I don't know if we need the separate function. And if we did, it should begin with an underscore since it's private, and moved in the private block

File: js/src/tooltip.js
Comment (Johann-S) [inline js/src/tooltip.js:44]: I would prefer a parameter called : `customClass` which would allow an array of string or a function it would be better if someone wants to add a lot of classes

File: js/src/tooltip.js
Comment (Johann-S) [reply js/src/tooltip.js:445]: yep it should be a private method and start with an `_`
btw it should return an array of classes and with we would be able to do just one call to `addClass` it should be better in terms of perfs

File: js/tests/unit/popover.js
Comment (Johann-S) [inline js/tests/unit/popover.js:70]: Can you assert the default classes are present too ?

File: js/tests/unit/tooltip.js
Comment (Johann-S) [inline js/tests/unit/tooltip.js:1297]: same here (checking if default classes are present)

File: js/tests/unit/tooltip.js
Comment (Johann-S) [inline js/tests/unit/tooltip.js:1323]: we run on old browsers in v4 and our unit tests doesn't use Babel, so now arrow function please

**Review by Johann-S — CHANGES_REQUESTED** [review]
Nice work @remeika! 👍 

Some changes to do but not so much

File: js/tests/unit/tooltip.js
Comment (XhmikosR) [reply js/tests/unit/tooltip.js:1323]: I'm going to try to enforce this in linting, I thought we already did.

File: js/src/tooltip.js
Comment (remeika) [reply js/src/tooltip.js:44]: @Johann-S  As this PR stands now, the additionalClasses property is a logic-less wrapper around [`jQuery.addClass()`](https://api.jquery.com/addClass/#addClass-classNames). As of jquery 3.3, addClass() does accept an array of classes. So my initial thinking was that in Bootstrap 4.X, we guarantee that a string or function are valid inputs (although passing in an array will work in environments with a modern version of jQuery), and then in Bootstrap 5 we could rely on a more modern version of jQuery. But I just checked `package.json` on the `main` branch, and jQuery is no longer listed as a peer dependency. Will Bootstrap 5 not depend on jQuery? If so I will definitely change my strategy.

File: js/src/tooltip.js
Comment (remeika) [reply js/src/tooltip.js]: Changed name to `_getCustomClass()`. See below for discussion about using built-in array processing in `$.addClass`.

File: js/tests/unit/popover.js
Comment (remeika) [reply js/tests/unit/popover.js]: 👍 done.

**Comment by cankitm** [comment]
> But we'd need a patch for the main branch anyway. About bundlewatch, leave that to us, we'll update the file before merging.

@XhmikosR @Johann-S  Can I pick the patch for main branch?

File: js/src/tooltip.js
Comment (Johann-S) [reply js/src/tooltip.js]: in v4 we can stay like you did, just the renaming (additionalClasses -> customClass) but since v5 is intended to work without jQuery I would prefer an array and a function

**Comment by XhmikosR** [comment]
My only concern is that people might ask for the ability add custom classes in all components. So, shouldn't we have a helper function and add this functionality in all components?

**Comment by Johann-S** [comment]
yup you're right it will be asked for sure.

But currently our code architecture doesn't allow to have one helper shared across our components because we don't use inheritance

**Comment by XhmikosR** [comment]
Maybe we should wait until this can be applied to all components? I feel like it's sort of incomplete if we don't add this for all components

**Comment by Johann-S** [comment]
it's a point of view, or we can add it when folks ask for it 🤷

**Comment by XhmikosR** [comment]
@remeika can you make a PR for main too please?

**Comment by XhmikosR** [comment]
@Johann-S LGTY?

@rohit2sharma95 if we merge this, can you make a PR against main later please? 🙂

**Comment by rohit2sharma95** [comment]
Okay sure 👍  @XhmikosR

**Review by Johann-S — APPROVED** [review]

**Pull request merged**
